elastic/elastic-agent
# Cloudnative ci automation

## What does this PR do?

This PR provides the relevant Jenkins CI automation to open Pull requests to kibana github repository in order to keep Cloud-Native teams manifests in sync with the manifests that are used into Fleet UI.

For full information check https://github.com/elastic/elastic-agent/issues/706



- Updated .ci/Jenkins file that is triggered upon PR requests of /elastic-agent/deploy/kubernetes/* changes
- Updated Makefile to add functionality needed to create the extra files for the new prs to kibana remote repository

## Why is it important?

Because syncing of Manifests was until now a manual job and was prone to errors.



## Checklist




## Author's Checklist


- [x] Jenkins File that is triggered only upon PR commit
- [x] Successful run of CI Jenkins https://fleet-ci.elastic.co/job/elastic-agent/job/elastic-agent-mbp/view/change-requests/job/PR-837/30/console
- [x] PR opened 
~~https://github.com/elastic/kibana/pull/138260~~ 
~~https://github.com/elastic/kibana/pull/138393~~
https://github.com/elastic/kibana/pull/138713

## How to test this PR locally

- Clone elastic-agent repo
- gh pr checkout 837
- Make a change under /deploy/kubernetes folder and you should see https://fleet-ci.elastic.co/job/elastic-agent/job/elastic-agent-mbp/view/change-requests/job/PR-837/ being triggered



Finally this PR need to be merged to master with https://github.com/elastic/elastic-agent/pull/837/files#diff-ef8b2f5e4d21d81c0eba4754dea3a3dfc2c67d26284f195165b650cf4aff7beaR246 updated

So we finally need to make sure that stage SyncK8s (the new stage we introduced) run only when:

1. expression { return env.K8S_CHANGES == "true" }   // If k8s changes happen under /deploy/kubernetes folder
2. only if there are changes between elast-agent and kibane manifests as checked here: https://github.com/elastic/elastic-agent/pull/837/files#diff-18ea46196e4e61745238c819aae128261e8c81b1025bc9e455092d3e9f22817bR55
3. only if PR is merged to master  from users from now on

## Related issues


- Closes #706

## Use cases



## Screenshots



## Logs
